component/src/test/java/org/wso2/extension/siddhi/map/text/sinkmapper/TextDefaultSinkMapperTestCase.java

```diff
@@ -469,4 +534,170 @@ public String getTopic() {
+        InputHandler stockStream = siddhiAppRuntime.getInputHandler("FooStream");
+
+        siddhiAppRuntime.start();
+
+        stockStream.send(new Object[]{"WSO2", 55.6f, 100L});
+        stockStream.send(new Object[]{"IBM", 57.678f, 100L});
+        stockStream.send(new Object[]{"GOOGLE", 50f, 100L});
+        stockStream.send(new Object[]{"REDHAT", 50f, 100L});
+        Thread.sleep(100);
+
+        ArrayList<String> symbolNames = new ArrayList<>();
+        symbolNames.add("WSO2.txt");
+        symbolNames.add("IBM.txt");
+        symbolNames.add("GOOGLE.txt");
+        symbolNames.add("REDHAT.txt");
+
+        File sink = new File(sinkUri);
+        if (sink.isDirectory()) {
+            for (File file : sink.listFiles()) {
+                if (symbolNames.contains(file.getName())) {
+                    count.incrementAndGet();
+                }
+            }
+            AssertJUnit.assertEquals(4, count.intValue());
+        } else {
+            AssertJUnit.fail(sinkUri + " is not a directory.");
+        }
+
+        Thread.sleep(1000);
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test
+    public void fileSinkTestGroup() throws InterruptedException {
+        log.info("test text default map with file io");
+        AtomicInteger count = new AtomicInteger();
+        ClassLoader classLoader = TextDefaultSinkMapperTestCase.class.getClassLoader();
+        String rootPath = classLoader.getResource("files").getFile();
+        String sinkUri = rootPath + "/sink";
+        String streams = "" +
+                "@App:name('TestSiddhiApp')" +
+                "define stream FooStream (symbol string, price float, volume long); " +
+                "@sink(type='file', @map(type='text',event.grouping.enabled='true'" +
+                "), append='false', " +
+                "file.uri='" + sinkUri + "/{{symbol}}.txt') " +
+                "define stream BarStream (symbol string, price float, volume long); ";
+
+        String query = "" +
+                "from FooStream " +
+                "select * " +
+                "insert into BarStream; ";
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(streams + query);
+        InputHandler stockStream = siddhiAppRuntime.getInputHandler("FooStream");
+
+        siddhiAppRuntime.start();
+
+        ArrayList<org.wso2.siddhi.core.event.Event> arrayList = new ArrayList<>(100);
+        for (int j = 0; j < 5; j++) {
+            arrayList.add(new org.wso2.siddhi.core.event
+                    .Event(System.currentTimeMillis(), new Object[]{"WSO2", 55.6f, 10}));
+            arrayList.add(new org.wso2.siddhi.core.event
+                    .Event(System.currentTimeMillis(), new Object[]{"IBM", 75.6f, 10}));
+        }
+        stockStream.send(arrayList.toArray(new org.wso2.siddhi.core.event.Event[10]));
+        Thread.sleep(100);
+
+        ArrayList<String> symbolNames = new ArrayList<>();
```

Comment:
Declaration should be a List

Reply:
Fixed in cdb7b86019f370bcb90c545043b2ad205e8e98df